source/plugin/Assets/GoogleMobileAds/Platforms/Unity/DummyAdBehaviour

```diff
@@ -0,0 +1,100 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using UnityEngine;
+using UnityEngine.UI;
+using GoogleMobileAds.Api;
+
+
+public class DummyAdBehaviour : MonoBehaviour
+{
+
+    public void AnchorAd(GameObject dummyAd, AdPosition position)
+    {
+        if (dummyAd != null) {
+            Image myImage = dummyAd.GetComponentInChildren<Image>();
+            RectTransform rect = myImage.GetComponent<RectTransform>();
+
+            float x = rect.sizeDelta.x/2;
+            float y = rect.sizeDelta.y/2;
+        
+            
+            switch (position)
+            {
+                case (AdPosition.TopLeft):
+                    rect.pivot = new Vector2(0.5f, 0.5f);
+                    rect.anchorMin = new Vector2(0, 1);
+                    rect.anchorMax = new Vector2(0, 1);
+                    rect.anchoredPosition = new Vector2(x, -y);
+                    break;
+                case (AdPosition.TopRight):
+                    rect.pivot = new Vector2(0.5f, 0.5f);
+                    rect.anchorMin = new Vector2(1, 1);
+                    rect.anchorMax = new Vector2(1, 1);
+                    rect.anchoredPosition = new Vector2(-x, -y);
+                    break;
+                case (AdPosition.Top):
+                    rect.pivot = new Vector2(0.5f, 0.5f);
+                    rect.anchorMin = new Vector2(0.5f, 1);
+                    rect.anchorMax = new Vector2(0.5f, 1);
+                    rect.anchoredPosition = new Vector2(0, -y);
+                    break;
+                case (AdPosition.Bottom):
+                    rect.pivot = new Vector2(0.5f, 0.5f);
+                    rect.anchorMin = new Vector2(0.5f, 0);
+                    rect.anchorMax = new Vector2(0.5f, 0);
+                    rect.anchoredPosition = new Vector2(0, y);
+                    break;
+                case (AdPosition.BottomRight):
+                    rect.pivot = new Vector2(0.5f, 0.5f);
+                    rect.anchorMin = new Vector2(1, 0);
+                    rect.anchorMax = new Vector2(1, 0);
+                    rect.anchoredPosition = new Vector2(-x, y);
+                    break;
+                case (AdPosition.BottomLeft):
+                    rect.pivot = new Vector2(0.5f, 0.5f);
+                    rect.anchorMin = new Vector2(0, 0);
+                    rect.anchorMax = new Vector2(0, 0);
+                    rect.anchoredPosition = new Vector2(x, y);
+                    break;
+                default:
+                    rect.pivot = new Vector2(0.5f, 0.5f);
+                    rect.anchorMin = new Vector2(0.5f, 0.5f);
+                    rect.anchorMax = new Vector2(0.5f, 0.5f);
+                    rect.anchoredPosition = new Vector2(0, 0);
+                    break;
+            }
+        } else {
+            Debug.Log("Invalid Dummy Ad");
```

Comment:
This is probably fine, but did you have alternative ideas for how to handle this besides logging?

Reply:
I could potentially put a placeholder in the same position that was requested but with no test ad image

Reply:
My vote would be to leave the log. What happens in iOS or Android if they pass in a size that doesn't exist?